apiserver/facades/client/controller/controller.go

```diff
@@ -172,14 +175,15 @@ func NewControllerAPI(
 		modelConfigServiceGetter:  modelConfigServiceGetter,
 		blockCommandServiceGetter: blockCommandServiceGetter,
 		proxyService:              proxyService,
-		controllerTag:             st.ControllerTag(),
 		modelExporter:             modelExporter,
 		store:                     store,
+		controllerUUID:            model.ControllerUUID(),
```

Comment:
This is wrong I believe. `model` here is coming from mongo. You can get the controller uuid from the arguments passed into the function.

Line 155 above is using the tag from the st model as well. Are we able to get the model uuid supplied to this function as well from the context in register?

Reply:
I would also suggest changing this type on the structure to be the controller tag and just make it once in this function. It is the only reason we need the value so might as well convert it to what we need now.

Reply:
thanks I must have missed this one. Find to leave it then IMO.

Reply:
np thanks tom!